src/components/molecules/PlanRoom/RoomTables/Table/index.tsx

```diff
@@ -0,0 +1,182 @@
+      );
+    },
+    [tables, setTables],
+  );
+
+  changeValueTableBorderRadius = useCallback(
+    (id: string, borderRadius: number) => {
+      setTables(
+        update(tables, {
+          [id]: {
+            $merge: { borderRadius },
+          },
+        }),
+      );
+    },
+    [tables, setTables],
+  );
+
+  addNewTable = (id: number) => {
+    tables[id] = { top: 0, left: 0, text: id.toString(), width: 100, height: 50, borderRadius: 0 };
+    let top = Math.floor(Math.random() * 100) + 40;
+    if (top > 80) {
+      top = 80;
+    }
+    moveTable((id - 1).toString(), Math.floor(Math.random() * 800) + 50, top);
+  };
+
+  const [, drop] = useDrop(
+    () => ({
+      accept: ItemTypes.TABLE,
+      drop(item: TypeItem, monitor) {
+        const delta = monitor.getDifferenceFromInitialOffset() as XYCoord;
+        let left = item.left + delta.x;
+        let top = item.top + delta.y;
+        if (top > 319) {
+          const difference: number = top - 319;
+          top -= difference;
+        }
+        if (top < 0) {
+          top = 0;
+        }
+        if (left < 0) {
+          left = 0;
+        }
+        moveTable(item.id, left, top);
+        return undefined;
+      },
+    }),
+    [moveTable],
+  );
+
+  save = () => {
+    console.log(tables);
+    return tables;
+  };
+
+  return (
+    <Box ref={drop} w="100%" h="100%" position="relative">
+      {Object.keys(tables).map((id) => {
+        const { left, top, text, height, width, borderRadius } = tables[id];
+        return (
+          <Table
+            key={id}
+            id={id}
+            left={left}
+            top={top}
+            height={height}
+            width={width}
+            borderRadius={borderRadius}
+            text={text}
+          >
+            {text}
+          </Table>
+        );
+      })}
+    </Box>
+  );
+};
+
+const Table = ({ id, left, top, height, width, borderRadius, text }: TableType) => {
+  const [{ isDragging }, drag] = useDrag(
+    () => ({
+      type: ItemTypes.TABLE,
+      item: { id, left, top },
+      collect: (monitor) => ({
+        isDragging: monitor.isDragging(),
+      }),
+    }),
+    [id, left, top],
+  );
+
+  if (isDragging) {
+    return <Box ref={drag} />;
+  }
+  return (
+    <Textarea
+      id={id}
+      ref={drag}
+      position="absolute"
+      borderRadius={borderRadius}
+      cursor="pointer"
+      textAlign="center"
+      left={left}
+      top={top}
+      width={width}
+      height={height}
+      backgroundColor={theme.colors.table[100]}
+      color="white"
+      maxLength={2}
+      focusBorderColor="none"
+      paddingTop="25px"
+      maxH="80px"
+      defaultValue={text}
+      onDoubleClick={() => {
+        changeValueTableBorderRadius(id.toString(), tableRadius(borderRadius));
+      }}
+      onChange={(event) => {
+        changeValueTable(id.toString(), event.target.value.toString());
+      }}
+    />
+  );
+};
+
+const tableRadius = (radius: number) => {
```

Comment:
This function creates code smell. You use magic numbers such as "40". Wouldn't it be easier, if you change the code and instead create function with name `getRadius` or more verbose `shouldProvideRadius`? You would only pass boolean value, and the responsibility of the function is to decide whether to add radius or not.

Example:
`const getRadius = (hasRadius: boolean) => hasRadius ? 40 : 0;`

Reply:
Well, yes :D I will use your idea to improve the notation of the function